src/gsy_e/models/strategy/area_agents/future_agent.py

```diff
@@ -0,0 +1,63 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of gsy_e.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from typing import TYPE_CHECKING
+
+from gsy_framework.constants_limits import ConstSettings
+
+from gsy_e.gsy_e_core.util import make_sa_name
+from gsy_e.models.market.future import FutureMarkets
+from gsy_e.models.strategy.area_agents.future_engine import FutureEngine
+from gsy_e.models.strategy.area_agents.two_sided_agent import TwoSidedAgent
+
+if TYPE_CHECKING:
+    from gsy_e.models.area import Area
+
+
+class FutureAgent(TwoSidedAgent):
+    """Handler for IAAEngines for the future markets."""
+
+    def __init__(self, *, owner: "Area", higher_market: FutureMarkets,
+                 lower_market: FutureMarkets,
+                 min_offer_age: int = ConstSettings.IAASettings.MIN_OFFER_AGE,
+                 min_bid_age: int = ConstSettings.IAASettings.MIN_BID_AGE):
+
+        super().__init__(owner=owner, higher_market=higher_market,
+                         lower_market=lower_market,
+                         min_offer_age=min_offer_age, min_bid_age=min_bid_age,
+                         do_create_engine=False)
+        self.engines = [
+            FutureEngine("High -> Low", higher_market, lower_market,
+                         min_offer_age, min_bid_age, self),
+            FutureEngine("Low -> High", lower_market, higher_market,
+                         min_offer_age, min_bid_age, self),
+        ]
+        self.name = make_sa_name(self.owner)
+
+    def _clean_up_engine_buffers(self) -> None:
```

Comment:
IMO this doesn't deserve its own method, it's only used once inside the `delete_engines` (It belongs there AFAIS)

Reply:
Ok, I can move the code to `delete_engines`